pkg/location/options.go

```diff
@@ -0,0 +1,79 @@
+package location
+
+import (
+	"context"
+	"io"
+)
+
+type options struct {
+	ctx     context.Context
+	stdin   io.Reader
+	stdout  io.Writer
+	parsers []parser
+}
+
+func makeOptions(opts ...Option) options {
+	opt := options{}
+	for _, o := range opts {
+		o(&opt)
+	}
+	return opt
+}
+
+type parser func(value string, opt options) (Reference, error)
+
+// Option is a functional option for location parsing.
+type Option func(*options)
+
+// WithDefaultTag sets the default tag that will be used if one is not provided.
```

Comment:
Sorry - missed this question.

In the Git parser there is one place where (if there is no branch/ref name given in the arg) it will reach out to the git repository to find the default branch name.

Because the git parsing calls through to the existing implementation, passing the context as an option enabled that behavior to be unchanged.